cmd/dm-ctl/main.go

```diff
@@ -31,37 +31,98 @@ import (
 	"github.com/pingcap/dm/pkg/utils"
 )
 
+func helpUsage(cfg *common.Config) {
+	fmt.Println("Usage: dmctl [global options] command [command options] [arguments...]")
+	fmt.Println()
+	ctl.Usage()
+	fmt.Println()
+	fmt.Println("Special Command:")
+	f := cfg.FlagSet.Lookup(common.EncryptCmdName)
+	fmt.Println(fmt.Sprintf("\t--%s %s", f.Name, f.Usage))
+	fmt.Println()
+	fmt.Println("Global Options:")
+	cfg.FlagSet.VisitAll(func(flag2 *flag.Flag) {
+		if flag2.Name == common.EncryptCmdName {
+			return
+		}
+		fmt.Println(fmt.Sprintf("\t--%s %s", flag2.Name, flag2.Usage))
+	})
+}
+
 func main() {
 	cfg := common.NewConfig()
-	err := cfg.Parse(os.Args[1:])
-	switch errors.Cause(err) {
-	case nil:
-	case flag.ErrHelp:
-		os.Exit(0)
-	default:
-		fmt.Printf("parse cmd flags err: %s", err)
-		os.Exit(2)
+	args := os.Args[1:]
+
+	// try to split one task operation from dmctl command
+	// because we allow user put task operation at last with two restrictions
+	// 1. one command one task operation
+	// 2. put task operation at last
+	cmdArgs := collectArgs(args)
+	lenArgs := len(args)
+	lenCmdArgs := len(cmdArgs)
+	if lenCmdArgs > 0 {
+		lenArgs = lenArgs - lenCmdArgs
 	}
 
-	utils.PrintInfo2("dmctl")
-	fmt.Println() // print a separater
+	if len(cmdArgs) > 0 {
+		commandMode(cmdArgs)
```

Comment:
when do we using _Global Options_?

Reply:
@3pointer ping

Reply:
Before subCommand?
`Usage: dmctl [global options] command [command options] [arguments...]`

Reply:
but in the current implementation, we ignore all global options?

Reply:
we can use global options, like `--V` `--master-addr`